### ISSUE TITLE
clang-tidy-16 cannot disable clang-analyzer-core checks

### ISSUE THREAD

**Issue by kiwixz** [llvm/llvm-project#61520]
It used to work with LLVM 15.

An example working as expected with another check:
```console
$ clang-tidy-16 --checks=* --list-checks | grep NewDelete          
    clang-analyzer-cplusplus.NewDelete
    clang-analyzer-cplusplus.NewDeleteLeaks
$ clang-tidy-16 --checks=*,-clang-analyzer-cplusplus.NewDelete --list-checks | grep NewDelete     
    clang-analyzer-cplusplus.NewDeleteLeaks
```

An example of a problematic check:
```console
$ clang-tidy-16 --checks=* --list-checks | grep CallAndMessage 
    clang-analyzer-core.CallAndMessage
    clang-analyzer-core.CallAndMessageModeling
$ clang-tidy-16 --checks=*,-clang-analyzer-core.CallAndMessage --list-checks | grep CallAndMessage
    clang-analyzer-core.CallAndMessage
    clang-analyzer-core.CallAndMessageModeling
```

**Comment by firewave** [comment]
Related to #59588 and #59589

**Comment by PiotrZSL** [comment]
To be honest that dependency on core checks is not needed at all.
Thing is that clang/include/clang/StaticAnalyzer/Checkers/Checkers.td define list of checks, their dependences and what are hidden and what are not. Clang-tidy just don't use this information and follow simple aproach.

**Comment by PiotrZSL** [comment]
May "help": https://reviews.llvm.org/D146396
But I wouldn't expect it to land into Clang 16, but free to do cherry-pick.

**Comment by kiwixz** [comment]
The thing is that it used to work with LLVM 15.  As I understand, clang-tidy force-enables the checks but filters the warnings at the end.  I think your patch will definitely improve speed by not running unneeded checks, but we still want to ensure the filter is working.

I narrowed the regression to a specific condition, it only happens when treating warnings as errors.

Submitted a fix here: https://reviews.llvm.org/D146520

**Comment by kiwixz** [comment]
Fixed with 9644368f974a095472832bbf9cd475ef5c2c307f
